feedthefox/users/providers/github/provider.py

```diff
@@ -0,0 +1,25 @@
+from allauth.account.models import EmailAddress
+from allauth.socialaccount import providers
+from allauth.socialaccount.providers.github.provider import GitHubProvider
+
+
+class FeedTheFoxGitHubProvider(GitHubProvider):
+    """Custom GitHub provider that handles a list of emails
+    from GitHub and adds them to the list of verified emails.
+    """
+    package = 'feedthefox.users.providers.github'
+
+    def get_default_scope(self):
+        """Use the "user:email" OAuth2 scope."""
+        return ['user:email']
+
+    def extract_email_addresses(self, data):
+        """Extract only the verified emails from GitHub."""
+        return [EmailAddress(email=entry['email'],
+                             verified=True,
+                             primary=entry['primary'])
+                for entry in data.get('email_addresses')
```

Comment:
Lets populate the list first and then return. This is getting a little bit obscure.